### DISCUSSION TITLE
"Debugging" / Management facilities in Equinox?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#153] [Q&A]
I recently though about that it would be cool to have more insight to OSGiFrameworks/applications running from my eclipse, for example like

- [The Knopflerfish Desktop](https://www.knopflerfish.org/desktop.html)
- [OSGi.fx](https://osgifx.com/)

and I came along https://www.eclipse.org/equinox/incubator/monitoring/index.php but it seems it is not part of equinox (or the equinox repo) so I wonder if this has ever come to life? Was it postponed? Is there maybe something similar already included yet?

Maybe https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.rest.html would be the today usage technique, but is this already something one can setup "easy" in equinox/eclipse and attach a nice UI to it?

**Comment by tjwatson** [comment]
AFAIK, this never made it out of the incubator and we don't have any release of it.  To make matters worse, I'm not sure where we could resurrect the source for this from the past.  Not that I think it would be all that useful to do so this much later though.

**Comment by HannesWell** [comment]
Isn't this question somehow similar to the suggestion to add a PDE spy to include the OSGi-fx Console?
https://github.com/eclipse-pde/eclipse.pde/issues/371
In the end to goal of both is to have better insides into the running OSGi app, isn't it?
